File: backend/Persistence/Models/Amenity.cs
using System;
using System.Collections.Generic;

namespace Persistence.Models;

public partial class Amenity
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<PropertyAmenity> PropertyAmenities { get; set; } = new List<PropertyAmenity>();
}

File: mobile/lib/src/features/dashboard/domain/entities/amenity.dart
import 'package:equatable/equatable.dart';

class Amenity extends Equatable {
  final String amenityName;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const Amenity({
    required this.amenityName,
    this.createdAt,
    this.updatedAt,
  });

  @override
  List<Object?> get props => [amenityName, createdAt, updatedAt];
}